contracts/PrisonersDilemma.sol

```diff
@@ -1,34 +1,53 @@
-pragma solidity ^0.4.25;
-
-//This contract mimics that of the prisoners dilemma
-//
-//+----------+----------------------+
-//|          |       Player 2       |
-//+----------+----------------------+
-//|          |       | Share | Take |
-//+          +-------+-------+------+
-//| Player 1 | Share | 1/1   | 5/0  |
-//+          +-------+-------+------+
-//|          | Take  | 0/5   | 0/0  |
-//+----------+-------+-------+------+
-//Rules:
-//- If two places 'share' they both receive 1 point and the game progresses
-//- If two players 'take', they receive no points and the game doesn't progress
-//- If either one of the players takes while the other player shares, the person who shared will receive no points
-//- Game ends when one of the players gets 20 points
-//- Ties are possible
+pragma solidity ^0.4.24;
+
+/**
+This contract mimics that of the prisoners dilemma
+
++----------+----------------------+
+|          |       Player 2       |
++----------+----------------------+
+|          |       | Share | Take |
++          +-------+-------+------+
+| Player 1 | Share | 1/1   | 5/0  |
++          +-------+-------+------+
+|          | Take  | 0/5   | 0/0  |
++----------+-------+-------+------+
+Rules:
+- If two places 'share' they both receive 1 point and the game progresses
+- If two players 'take', they receive no points and the game doesn't progress
+- If either one of the players takes while the other player shares, the person who shared will receive no points
+- Game ends when one of the players gets 20 points
+- Ties are possible
+**/
 
 contract PrisonersDilemma {
 
-    //Variables for player's addresses
-    //Variables for player's scores
-    //Variable for winner of contract
+    enum ActionChoices { NoChoice, Share, Take }
+
+    struct Player {
+        address addr;
+        ActionChoices choice;
+        uint score;
+    }
+
+    //State Variables
+    mapping (address => Player) public players;
+    address public winner = address(0); //empty address
+
+    //Events
+    event ContractInitialized(address _player1, address _player2);
+    event PlayerSelectedChoice(address _player);
+    event AlertWinner(address _player);
+
+    constructor(address _player1, address _player2) public {
+
```

Comment:
I'm pretty sure these should be memory since  we cannot create storage variables inside a function. If we tried it'd otherwise be a storage pointer and we might modify the wrong variable.